src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -172,49 +300,273 @@ def selectedshapechange(self):
         self.communicator.statusBarUpdateSignal.emit(
             "The option Ellipsoid has not been implemented yet.")
 
-    def loadFile(self):
+    def toggle_error_functionality(self, verificationMsg=None):
+        """Disables/Enables some functionality if the state of the GUI means calculation cannot proceed
+
+        This function is called during any process whenever there is a risk that the state
+        of the GUI will make the data invalid for plotting, drawing or saving. If that is the 
+        case then this functionality is disabled. This function is currently called when two
+        files are being verified for compatibility, and when textboxes enter 'intermediate' states.
+        It also optionally changes the validation error message for the user.
+
+        :param msg: The error message due to file verification which should be displayed
+            to the user on the GUI. Defaults to `None` in which case the error message is
+            not changed.
+        :type msg: str or None
+        """
+        verificationEnable = self.verified or not (self.is_mag and self.is_nuc)
+        lineEditsEnable = len(self.invalidLineEdits) == 0
+        # disable necessary buttons to prevent the attempted merging of incompatible files
+        self.cmdDraw.setEnabled(verificationEnable and lineEditsEnable)
+        self.cmdDrawpoints.setEnabled(verificationEnable and lineEditsEnable)
+        self.cmdSave.setEnabled(verificationEnable and lineEditsEnable)
+        self.cmdCompute.setEnabled(verificationEnable and lineEditsEnable)
+        # alter the error message if a new message is provided
+        # verification is only carried out once so if msg=None do not set the msg to ""
+        # but simply don't alter it - this means the message is preserved and re-verification
+        # s not called when the files have not been changed
+        if verificationMsg is not None:
+            self.lblVerifyError.setText('<font color="#FF0000">' + verificationMsg + '</font>')
+        # display the message
+        self.lblVerifyError.setVisible(not verificationEnable)
+
+    def verify_files_match(self):
+        """Verifies that enabled files are compatible and can be combined
+
+        When the user wishes to combine two different files for nuclear and magnetic
+        data they must have the same 3D data points in real-space. This function
+        decides whther verification of this is necessary and if so carries it out.
+        In the case that the two files have the same real-space data points in different
+        orders this function re-orders the stored data within the MagSLD objects to make
+        them align. The full verification is only carried out once for any pair of loaded
+        files.
+        """
```

Comment:
Move what code that doesn't depend on gui to sascalc. You may want to mix and match files when running a remote batch job for example.

Reply:
This issue will be addressed with [PR1898](https://github.com/SasView/sasview/pull/1898).